### PR TITLE
Core/Spells: Define and implement some new spell attributes

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Tests performed:** Compiles

Any feedback is welcomed

### CODE REVIEW

**Comment by tkrokli** [comment]
Cosmetic detail: if `SPELL_ATTR3_NO_PUSHBACK` gets approved, consider improving its comment (25) text ;-)

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -17847,7 +17847,7 @@ void Player::_LoadAuras(PreparedQueryResult result, uint32 timediff)
             }
 
             // negative effects should continue counting down after logout
-            if (remaintime != -1 && !spellInfo->IsPositive())
+            if (remaintime != -1 && ((!spellInfo->IsPositive() && spellInfo->Id != 15007) || spellInfo->HasAttribute(SPELL_ATTR4_FADES_WHILE_LOGGED_OUT))) // Resurrection sickness should not fade while logged out
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: Uh, I think it should fade when logged out but let me test it on retail first

Comment (xinef1) [inline src/server/game/Entities/Player/Player.cpp]: I'm 99% sure it didn't fade on wotlk, but ill adjust the code if necessary :)

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 2]
```diff
@@ -519,9 +519,9 @@ enum SpellAttr6
     SPELL_ATTR6_CAN_TARGET_UNTARGETABLE          = 0x01000000, // 24
     SPELL_ATTR6_NOT_RESET_SWING_IF_INSTANT       = 0x02000000, // 25 Exorcism, Flash of Light
     SPELL_ATTR6_UNK26                            = 0x04000000, // 26 related to player castable positive buff
-    SPELL_ATTR6_UNK27                            = 0x08000000, // 27
+    SPELL_ATTR6_LIMIT_PCT_HEALING_MODS           = 0x08000000, // 27 some custom rules - complicated
```
Comment (ariel-) [inline src/server/game/Miscellaneous/SharedDefines.h]: Defined but not implemented

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (Aokromes) [inline src/server/game/Entities/Player/Player.cpp]: At least on legion Resurrection sickness fades while logged out.

Comment (Aokromes) [inline src/server/game/Entities/Player/Player.cpp]: https://us.battle.net/forums/en/wow/topic/1670088555
https://us.battle.net/forums/en/wow/topic/2228173818
https://us.battle.net/forums/en/wow/topic/2369682005

so, @xinef1 is right

Comment (ariel-) [inline src/server/game/Entities/Player/Player.cpp]: Better to apply spellinfo correction removing the attribute?

Comment (r00ty-tc) [reply src/server/game/Entities/Player/Player.cpp]: The issue isn't spellinfo from what I can see. It's a negative spell and the check for not positive traps it. The spell doesn't have the fades flag set and this is correct.

The question we need to ask is whether all the spells that have the flag are the correct list and if so, then we can drop the positive check. If not, then we keep in the exception.

**Review by Shauren — APPROVED** [review]

**Pull request merged**